src/super_gradients/training/utils/media/image.py

```diff
@@ -77,12 +96,26 @@ def load_image(image: ImageSource) -> np.ndarray:
         - str:              A string representing either a local file path or a URL to an image
 
     :param image: Single image of supported types.
-    :return:      Image as numpy arrays. If loaded from string, the image will be returned as RGB.
+    :param input_image_channels: Number of channels that model expects as input.
+                                 This value helps to infer the layout of the input image array.
+                                 As of now this argument has default value of 3, but in future it will become mandatory.
```

Comment:
Why? 3 is a pretty good default choice IMO.

Reply:
This is a good default value, however I'm convinced that load_image should not hold any assumption on number of channels. In fact I'd rather require data layout to be passed explicitly, instead of trying to infer it based on number of channels. This is big design change, so I can live without it for now.

Anyhow, my point is Pipeline class should be knowing what input channels should be (It has all the information needed to infer this from model instance) and propagate this value to `load_image`.
So the idea from removing the default value then is to enforce this parameter is explicitly passed.